### PR TITLE
feat(settings): Add first Page of the settings dialogs

### PR DESCRIPTION
**Pull request by Lordnoname**
feat(setting-dialog): Implement Settingsdialog into the UserInterface

feat(school): Add school identifier and toString method

feat(school): Add school identifier

feat(restart): Add restart thread -> not tested yet

feat(school, settings): Add the possibility to change between schools

### CODE REVIEW

File: src/main/java/de/thielpaul/vertretungsplan/thread/RestartThread.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      //Locates .jar File
+      final File jarFile =
+          new File(mainInstance.getClass().getProtectionDomain().getCodeSource().getLocation()
+              .toURI());
+      // Checks if this is a .jar file
+      // Returns if it isnt a .jar file -> it's a .class file
+      if (jarFile.getName().endsWith(".jar") == false) return;
+
+      //Executes command to run a new instance of the programm
+      Runtime.getRuntime().exec(new String[] {javaBin, "-jar", jarFile.getPath()});
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/thread/RestartThread.java]: Why starting the Application in a new Java Virtual Machine? You can easily start a Non-Daemon Process which bootstraps the `main` method instead.

**Comment by Lordnoname** [comment]
Bootstrapping the `main` method isn't a clean restart of the Application. The only way to trash the JVM completely is to stop the current process and `starting a clean new one`.

**Comment by DevVersion** [comment]
@Lordnoname That's a completely wrong statement. There are several ways to clear the JVM, for example you can close the Application as normal as every Application and then you request the Garbage Collection to collect unreferenced  objects / variables and clear them.

Then you can bootstrap the Application through calling the `main` Method with the predefined `bootstrap` arguments. That will be a clean restart, without charging the `system runtime`.

**Comment by DevVersion** [comment]
Due, this PR is currently working, I will merge that, but I expect you, to use my suggestion after this PR is merged.

**Pull request closed without merging**